drivers/mspi/mspi_nrfe.c

```diff
@@ -0,0 +1,644 @@
+#endif
+	while (!atomic_test_and_clear_bit(&ipc_atomic_sem, opcode)) {
+#if defined(CONFIG_SYS_CLOCK_EXISTS)
+		if ((k_uptime_get_32() - start) > timeout) {
+			return -ETIMEDOUT;
+		};
+#else
+		repeat--;
+		if (!repeat) {
+			return -ETIMEDOUT;
+		};
+#endif
+		k_sleep(K_USEC(1));
+	}
+
+	return 0;
+}
+
+/**
+ * @brief Send a data struct to the FLPR core using the IPC service.
+ *
+ * The function sends a data structure to the FLPR core,
+ * inserting a byte at the beginning responsible for the opcode.
+ *
+ * @param opcode The NRFE MSPI opcode.
+ * @param data The data to send.
+ * @param len The length of the data to send.
+ *
+ * @return 0 on success, negative errno code on failure.
+ */
+static int send_with_opcode(enum nrfe_mspi_opcode opcode, const void *data, size_t len)
+{
+	uint8_t buffer[len + 1];
+
+	buffer[0] = (uint8_t)opcode;
+	memcpy(&buffer[1], data, len);
+
+	return mspi_ipc_data_send(opcode, buffer, sizeof(buffer));
+}
+
+/**
+ * @brief Send a configuration struct to the FLPR core using the IPC service.
+ *
+ * @param opcode The configuration packet opcode to send.
+ * @param config The data to send.
+ * @param len The length of the data to send.
+ *
+ * @return 0 on success, negative errno code on failure.
+ */
+static int send_config(enum nrfe_mspi_opcode opcode, const void *config, size_t len)
+{
+	int rc;
+
+	rc = send_with_opcode(opcode, config, len);
+	if (rc < 0) {
+		LOG_ERR("%d: Configuration send failed: %d", __LINE__, rc);
+		return rc;
+	}
+
+	rc = nrfe_mspi_wait_for_response(opcode, CONFIG_TIMEOUT_MS);
+	if (rc < 0) {
+		LOG_ERR("%d: FLPR config response timeout: %d!", __LINE__, rc);
+	}
+
+	return rc;
+}
+
+/**
+ * @brief Configures the MSPI controller based on the provided spec.
+ *
+ * This function configures the MSPI controller according to the provided
+ * spec. It checks if the spec is valid and sends the configuration to
+ * the FLPR.
+ *
+ * @param spec The MSPI spec to use for configuration.
+ *
+ * @return 0 on success, negative errno code on failure.
+ */
+static int api_config(const struct mspi_dt_spec *spec)
+{
+	const struct mspi_cfg *config = &spec->config;
+	const struct mspi_nrfe_config *drv_cfg = spec->bus->config;
+	const struct gpio_dt_spec *ce_gpio;
+
+	if (config->op_mode != MSPI_OP_MODE_CONTROLLER) {
+		LOG_ERR("%u, only support MSPI controller mode.", __LINE__);
+		return -ENOTSUP;
+	}
+
+	if (config->dqs_support) {
+		LOG_ERR("%u, only support non-DQS mode.", __LINE__);
+		return -ENOTSUP;
+	}
+
+	if (config->max_freq > drv_cfg->mspicfg.max_freq) {
+		LOG_ERR("%u, max_freq too large.", __LINE__);
+		return -ENOTSUP;
+	}
+
+	for (ce_gpio = config->ce_group; ce_gpio < &config->ce_group[config->num_ce_gpios];
+	     ce_gpio++) {
+		if (ce_gpio->pin < NRFE_MSPI_MIN_CE_PIN_NUMBER) {
+			LOG_ERR("%u, wrong GPIO pin %d number! Needs to be >= %d", __LINE__,
+				ce_gpio->pin, NRFE_MSPI_MIN_CE_PIN_NUMBER);
+			return -ENODEV;
+		}
+	}
+
+	if (config->re_init) {
```

Comment:
This will only apply configuration if `re_init` is enabled, is that expected?

Reply:
We don't have a controller reinitialization routine, so I changed the code so that the configuration is always sent to FLPR.